ArcGISRuntimeSDKQt_CppSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.cpp

```diff
@@ -0,0 +1,368 @@
+    {
+      m_valueSelectionListModel.clear();
+      const CodedValueDomain codedValueDomain = static_cast<CodedValueDomain>(domain);
+
+      for (CodedValue codedValue: codedValueDomain.codedValues())
+        m_valueSelectionListModel.append(codedValue.name());
+
+      m_textFieldVisible = false;
+    }
+    else
+    {
+      m_textFieldVisible = true;
+    }
+    emit valueSelectionListModelChanged();
+    emit textFieldVisibleChanged();
+  }
+}
+
+void ConfigureSubnetworkTrace::addCondition(const QString& selectedAttribute, int selectedOperator, const QVariant& selectedValue)
+{
+  // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+
+  if (!m_traceConfiguration)
+    m_traceConfiguration = new UtilityTraceConfiguration(this);
+
+  UtilityTraversability* traversability = m_traceConfiguration->traversability();
+  if (!traversability)
+    m_traceConfiguration->setTraversability(new UtilityTraversability(this));
+
+  UtilityNetworkAttribute* selectedNetworkAttribute = m_networkDefinition->networkAttribute(selectedAttribute);
+
+  const QVariant convertedSelectedValue = convertToDataType(selectedValue, selectedNetworkAttribute->dataType());
+
+  const UtilityAttributeComparisonOperator selectedOperatorEnum = static_cast<UtilityAttributeComparisonOperator>(selectedOperator);
+
+
+  // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+  UtilityTraceConditionalExpression* expression = new UtilityNetworkAttributeComparison(selectedNetworkAttribute, selectedOperatorEnum, convertedSelectedValue, this);
+
+  if (traversability->barriers())
+  {
+    UtilityTraceConditionalExpression* otherExpression = static_cast<UtilityTraceConditionalExpression*>(m_traceConfiguration->traversability()->barriers());
+
+    // NOTE: You may also combine expressions with UtilityTraceAndCondition
+    UtilityTraceConditionalExpression* combineExpressions = new UtilityTraceOrCondition(otherExpression, expression, this);
+
+    m_expressionBuilder = expressionToString(combineExpressions);
+    emit expressionBuilderChanged();
+
+    m_traceConfiguration->traversability()->setBarriers(combineExpressions);
+  }
+  else
+  {
+    m_expressionBuilder = expressionToString(expression);
+    emit expressionBuilderChanged();
+
+    m_traceConfiguration->traversability()->setBarriers(expression);
+  }
+}
+
+void ConfigureSubnetworkTrace::changeIncludeBarriersState(bool includeBarriers)
+{
+  m_traceConfiguration->setIncludeBarriers(includeBarriers);
+}
+
+void ConfigureSubnetworkTrace::changeIncludeContainersState(bool includeContainers)
+{
+  m_traceConfiguration->setIncludeContainers(includeContainers);
+}
+
+void ConfigureSubnetworkTrace::reset()
+{
+  // Reset the barrier condition to the initial value.
+  m_traceConfiguration->traversability()->setBarriers(m_initialExpression);
```

Comment:
There is code above that checks if `m_traceConfiguration->traversability()` is valid. Can we assume here that it is always valid?

Reply:
Yes `m_traceConfiguration` and `m_traceConfiguration->traversability()` is expected to always be valid. These can both be removed from `addCondition` for both C++ and QML.